src/main/groovy/com/ullink/NuGetSpec.groovy

```diff
@@ -65,24 +61,44 @@ class NuGetSpec extends Exec {
     }
 
     String supplementDefaultValueOnNuspec(String nuspecString) {
+        def final msbuildTaskExists = project.tasks.findByName('msbuild') != null
 
         def root = new XmlSlurper(false, false).parseText(nuspecString)
 
         def defaultValues = {}
-        def applyDefaultValue = { String node, String value ->
+        def setDefaultMetadata = { String node, String value ->
             if (root.metadata[node].isEmpty()) {
                 defaultValues <<= { delegate."$node" value }
             }
         }
-        applyDefaultValue ('id', project.name)
-        applyDefaultValue ('version', project.version)
-        applyDefaultValue ('description', project.description)
-
-        if (root.metadata.isEmpty()) {
-            root.appendNode { metadata defaultValues }
-        } else {
-            root.metadata.appendNode defaultValues
+
+        setDefaultMetadata ('id', project.name)
+        setDefaultMetadata ('title', project.name)
+        setDefaultMetadata ('version', project.version)
+        setDefaultMetadata ('description', project.description ? project.description : project.name)
+
+        def appendAndCreateParentIfNeeded = {
+            parentNodeName, children ->
+                if (root."$parentNodeName".isEmpty()) {
+                    root.appendNode { "$parentNodeName" children }
+                } else {
+                    root."$parentNodeName".appendNode children
+                }
         }
+
+        appendAndCreateParentIfNeeded('metadata', defaultValues)
+
+        if (msbuildTaskExists) {
+            def defaultFiles = {}
+            project.msbuild.mainProject.dotnetArtifacts.each {
+                artifact ->
+                    def fwkFolderVersion = project.msbuild.mainProject.properties.TargetFrameworkVersion.toString().replace('v', '').replace('.', '')
+                    defaultFiles <<= { file(src: artifact.toString(), target: 'lib/net' + fwkFolderVersion) }
+            }
+
+            appendAndCreateParentIfNeeded('files', defaultFiles)
```

Comment:
I see few options.
- If files are already provided we don't add them.
- We define a "msbuildArtifactsAsNuspecFiles" in the nuspec task (false by default). If true, we add them.
- The same as the first but we define a "forceMsbuildArtifactsAsNuspecFiles" for those that want them to be added anyway.

I think the first one make more sense.